<?xml version="1.0" encoding="UTF-8"?>
<root BTCPP_format="4"
      main_tree_to_execute="main">
  <BehaviorTree ID="main">
    <Sequence>
      <!-- <SwipeTail num_swipes="5"/> -->
      <MoveToPose posX="0.33"
                  posY="0.33"
                  theta="45"
                  threshold="0.025"
                  timeout="5000"/>
      <MoveToPose posX="0.45"
                  posY="0.2"
                  theta="0"
                  threshold="0.025"
                  timeout="10000"/>
      <MoveToPose posX="0.9"
                  posY="0.2"
                  theta="-5"
                  threshold="0.025"
                  timeout="10000"/>
      <MoveToPose posX="2.75"
                  posY="0.3"
                  theta="-5"
                  threshold="0.025"
                  timeout="10000"/>
      <MoveToPose posX="3.00"
                  posY="0.45"
                  theta="45"
                  threshold="0.025"
                  timeout="10000"/>
      <MoveToPose posX="3.25"
                  posY="1.2"
                  theta="90"
                  threshold="0.025"
                  timeout="10000"/>
      <Logging message="Done"/>
      <KeepRunningUntilFailure>
        <CheckForRestart/>
      </KeepRunningUntilFailure>
    </Sequence>
  </BehaviorTree>

  <BehaviorTree ID="odomTest">
    <Sequence>
      <!-- <SwipeTail num_swipes="5"/> -->
      <MoveToPose posX="0"
                  posY="1"
                  theta="0"
                  threshold="0.025"
                  timeout="5000"/>
      <KeepRunningUntilFailure>
        <CheckForRestart/>
      </KeepRunningUntilFailure>
    </Sequence>
  </BehaviorTree>

  <!-- Description of Node Models (used by Groot) -->
  <TreeNodesModel>
    <Action ID="CheckForRestart"
            editable="true"/>
    <Action ID="MoveToPose"
            editable="true">
      <input_port name="posX">m</input_port>
      <input_port name="posY">m</input_port>
      <input_port name="theta">deg</input_port>
      <input_port name="threshold"
                  default="0.025">m</input_port>
      <input_port name="timeout"
                  default="5000">ms</input_port>
    </Action>
    <Action ID="SwipeTail"
            editable="true">
      <input_port name="num_swipes"/>
    </Action>
  </TreeNodesModel>

</root>
